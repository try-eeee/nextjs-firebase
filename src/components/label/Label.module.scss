.label {
  display: block;
  margin-bottom: 0.5rem;
  font-size: 1rem;
  color: #333;
}

.left {
  text-align: left;
}

.center {
  text-align: center;
}

.right {
  text-align: right;
}

.required {
  background-color: #f8d7da;
  color: #721c24;
  padding: 0.2rem 0.5rem;
  border-radius: 1rem;
  font-size: 0.75rem;
  margin-left: 0.5rem;
  display: inline-block;
}
