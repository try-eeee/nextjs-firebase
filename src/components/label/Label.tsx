"use client";

import styles from "./Label.module.scss";

interface LabelProps {
  htmlFor: string;
  children: React.ReactNode;
  align: "left" | "center" | "right";
  required?: boolean;
}

export function Label(props: LabelProps) {
  const { htmlFor, children, align, required } = props;
  return (
    <label htmlFor={htmlFor} className={`${styles.label} ${styles[align]}`}>
      {children}
      {required && <span className={styles.required}>必須</span>}
    </label>
  );
}
