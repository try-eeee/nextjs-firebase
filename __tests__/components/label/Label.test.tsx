import React from "react";
import { render } from "@testing-library/react";
import { Label } from "@/components/label/Label";

describe("Label UnitTest", () => {
  test("Snapshot Test Of Label", () => {
    const { asFragment } = render(
      <Label htmlFor="lastName" align="center" required>
        å§“
      </Label>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
