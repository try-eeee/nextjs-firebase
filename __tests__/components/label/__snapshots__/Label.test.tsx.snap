// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label UnitTest Snapshot Test Of Label 1`] = `
<DocumentFragment>
  <label
    class="label center"
    for="lastName"
  >
    姓
    <span
      class="required"
    >
      必須
    </span>
  </label>
</DocumentFragment>
`;
